### ISSUE TITLE
Getting the direction of scroll in Scrolled event

### ISSUE THREAD

**Issue by iqbalansari** [alexrainman/CarouselView#313]
Hi Alex,

First of thanks for the fantastic work with this plugin, it has been super useful on the project I am currently working on.

I was wondering if it would be possible to add direction to the parameters of the Scrolled event that was introduced in v5.1.2. 

My use-case is that I want to move some things on screen as the carousel swipe changes (something like tab indicator in Android's default tab), the Scrolled event in the current form is not suitable for this purpose  since it just gives a percent and not the direction of the scroll, however that alone cannot be used to decide if the thing should move forward or backward.

Let me know if I am missing something. I might be able to create a pull request with the feature if you think it is worthwhile including it in the library.

Thanks

**Comment by alexrainman** [comment]
You can easily get the direction of the scroll tracking the previous position and the new one when when position has been selected.

**Comment by iqbalansari** [comment]
First of all, sorry for the delayed response. 

I initially tried the approach you mention, Unfortunately this does not work, because the scroll percentage is always increasing. 

For example suppose I have two views in the carousel (A and B). While swiping from A to B I get the following percentage in scroll events

0, 1, 5, 8, 9, ... 60, 74

While swiping **back** from B and A I get

0, 1, 7, 8, ... 59, 65, 70

Basically in both the cases the value is always increasing making it impossible to detect the direction of the scroll

**Comment by alexrainman** [comment]
I am talking about carousel position that you receive when PositionSelected is called.

**Comment by jelle-vdst** [comment]
I have a tab control like you describe you want to make (not allowed to share code tho).

Like alex also suggested I use the carousel position to see what tab should be selected. 
In the PositionSelected  event, you check what page you are on and change the selected tab that corresponds with that page.

**Comment by iqbalansari** [comment]
@alexrainman @jelle-vdst Thanks for the suggestion, however the approach that you describe would work only when you do not need do any processing while the tabs are changing. I want to move the slider while the user is swiping between tabs (not after the tabs have changed), I need to implement something like the following (the demo is taken from react-native-tab-view's readme). As you can see the slider moves even while the user is swiping through the views


![](https://raw.githubusercontent.com/satya164/react-native-tab-view/master/demo/demo.gif)

**Comment by jelle-vdst** [comment]
Have you already tried adding gesture recognizer of your own on top of the carousel to detect direction?

**Comment by iqbalansari** [comment]
Not yet I have reserved that as the last option, if all else fails. I think it is a reasonable expectation that the carousel view provides the direction along with the scroll percentage, especially because without the direction the scroll percent is somewhat meaningless.

**Comment by alexrainman** [comment]
MR.Gestures can be used for that

**Comment by alexrainman** [comment]
Reporting % has been tricky already. I can explore if it is possible to add Direction.

**Comment by alexrainman** [comment]
I think i can do it. iOS will be tricky but it is possible. Wait for next release later today or tomorrow. I need to ship another fix though.

**Comment by iqbalansari** [comment]
Thanks a lot @alexrainman, please let me know if I can help with implementing this, I really appreciate the hardwork you have put into this plugin.

**Comment by iqbalansari** [comment]
Would you mind a pull request implementing this feature, I have a rough idea of how this can be added

**Comment by alexrainman** [comment]
It is almost done. Thanks

**Comment by alexrainman** [comment]
Will release today.
